sklearn/tests/test_kernel_approximation.py

```diff
@@ -114,34 +114,41 @@ def test_additive_chi2_sampler():
     Y_neg[0, 0] = -1
     msg = "Negative values in data passed to"
     with pytest.raises(ValueError, match=msg):
-        transform.transform(Y_neg)
+        transformer.fit(Y_neg)
+
+    # test that the sample interval is set correctly
+    sample_intervals = [0.8, 0.5, 0.4]
+    for sample_interval in sample_intervals:
+        # test that the sample_interval is initialized correctly
+        transformer = AdditiveChi2Sampler(sample_interval=sample_interval)
+        transformer.fit(X)
+        assert transformer.sample_interval == sample_interval
 
     # test error on invalid sample_steps
-    transform = AdditiveChi2Sampler(sample_steps=4)
+    transformer = AdditiveChi2Sampler(sample_steps=4)
     msg = re.escape(
         "If sample_steps is not in [1, 2, 3], you need to provide sample_interval"
     )
     with pytest.raises(ValueError, match=msg):
-        transform.fit(X)
-
-    # test that the sample interval is set correctly
-    sample_steps_available = [1, 2, 3]
-    for sample_steps in sample_steps_available:
-
-        # test that the sample_interval is initialized correctly
-        transform = AdditiveChi2Sampler(sample_steps=sample_steps)
-        assert transform.sample_interval is None
-
-        # test that the sample_interval is changed in the fit method
-        transform.fit(X)
-        assert transform.sample_interval_ is not None
+        transformer.fit(X)
```

Comment:
let's refactor slightly the test by isolating this failure in a specific test function.
We probably want to call `fit_transform` and `transform` since it would be raised in both case (parametrizing with pytest would do the job).